solr/core/src/java/org/apache/solr/search/facet/SlotAcc.java

```diff
@@ -394,7 +394,16 @@ public CountSlotAcc(FacetContext fcontext) {
   int[] result;
   public CountSlotArrAcc(FacetContext fcontext, int numSlots) {
     super(fcontext);
-    result = new int[numSlots];
+    
+    String key = fcontext.level + this.getClass().getSimpleName();
+    result = (int[]) fcontext.getReusable(key);
+    if (result == null || result.length < numSlots) {
+      result = new int[numSlots];
+      fcontext.addReusable(key, result);
```

Comment:
If null != result is it reasonable to reset the value under the key to the newly created array?